### PR TITLE
Added check for spaces in filename on upload

### PR DESCRIPTION
**Pull request by crocodoyle**
Reorganized the upload-dialog javascript because there were many functions that set the disabled status of the "Upload" button, which produced unreliable behaviour.

Fix for #365 .

### CODE REVIEW

File: BrainPortal/public/javascripts/userfiles.js [hunk 1]
```diff
@@ -722,25 +722,45 @@ $(function() {
...
+            var warning_text = "";
             var max = parseInt($('#upload-dialog').data('max-upload-size'));
+            var bad_file;
+            var selected = !!$(this).val();
 
-            var visible = (
-              this.files && this.files[0] &&
-              max && this.files[0].size > max
-            );
+            var spaces_in_filename = $(this).val().includes(" ");
```
Comment (prioux) [inline BrainPortal/public/javascripts/userfiles.js]: Actually, why not compare it to the regex we use in the userfile model's validations ? That way we can complain about other illegal characters.

File: BrainPortal/public/javascripts/userfiles.js [hunk 2]
```diff
@@ -722,25 +722,45 @@ $(function() {
...
+            if ( max > 0 ){
+              file_too_big = this.files && this.files[0] && max && this.files[0].size > max
+            } else {
+              file_too_big = false;
+            }
+
+            bad_file = ( spaces_in_filename || file_too_big );
+
+            if ( spaces_in_filename ) {
+              warning_text += "Spaces in filename not allowed! ";
```
Comment (prioux) [inline BrainPortal/public/javascripts/userfiles.js]: Then this message would change to "Illegal characters in filename; filenames can only contain so and so".

File: BrainPortal/public/javascripts/userfiles.js [hunk 3]
```diff
@@ -773,15 +793,6 @@ $(function() {
             $('#up-type').val(data).trigger('change.uf');
           });
         })
```
Comment (prioux) [inline BrainPortal/public/javascripts/userfiles.js]: You sure that this wasn't needed ?

**Comment by prioux** [comment]
Andrew, I'm closing the pull request, for some reason it now integrates all sort of changes from other unrelated pull requests.

Can you make a new one? Make sure you start from a freshly synchronized 'dev' in your work directory.

**Comment by prioux** [comment]
I think it's normal. So just carry on with the comments...

File: BrainPortal/public/javascripts/userfiles.js (hunk 3 above)
Comment (crocodoyle) [inline BrainPortal/public/javascripts/userfiles.js]: It's still there, I just moved everything that modified the "disabled" status of the upload button into one method so they don't undo the work of the other ones

File: BrainPortal/public/javascripts/userfiles.js (hunk 2 above)
Comment (crocodoyle) [inline BrainPortal/public/javascripts/userfiles.js]: I have tried to use the userfiles model check for file validity, but I don't understand this RegEx:
`  def self.is_legal_filename?(basename)
    return true if basename && basename.match(/^[a-zA-Z0-9][\w\~\!\@\#\%\^\&\*\(\)\-\+\=\:\[\]\{\}\|\<\>\,\.\?]*$/)
    false
  end`

It appears to allow spaces?

**Comment by natacha-beck** [comment]
@prioux and @crocodoyle I take a look to this PR and it's seems to have some extra stuff from others PR (everything related to test in bourreau side) like Pierre said. Can you take a second look.

**Comment by crocodoyle** [comment]
It's because I pulled some things that were merged to dev and then committed them to this branch instead of rebasing.

**Comment by prioux** [comment]
Can you try rebasing this to skip the first three commits by me?

**Comment by prioux** [comment]
Or even better, rebase this to the current HEAD of the aces/dev branch .

**Comment by crocodoyle** [comment]
I think it's good now

**Comment by prioux** [comment]
All good Andrew!

**Pull request merged**